dotnet/src/Microsoft.AutoGen/Core/IAgentRuntime.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// IAgentRuntime.cs
+
+using Google.Protobuf;
+using Microsoft.AutoGen.Contracts;
+namespace Microsoft.AutoGen.Core;
+
+public interface IAgentRuntime
+{
+    IServiceProvider RuntimeServiceProvider { get; }
+    ValueTask RuntimeSendRequestAsync(Agent agent, RpcRequest request, CancellationToken cancellationToken = default);
+    ValueTask RuntimeSendResponseAsync(RpcResponse response, CancellationToken cancellationToken = default);
```

Comment:
From the agent's perspective of the runtime this should be:
```suggestion
    ValueTask<RpcResponse> RuntimeSendRequestAsync(Agent agent, RpcRequest request, CancellationToken cancellationToken = default);
```

Reply:
trying to keep the pr smaller - coming in a separate one.